mifosng-android/src/main/java/com/mifos/mifosxdroid/online/datatable/DataTableFragment.kt

```diff
@@ -4,140 +4,63 @@ import android.os.Bundle
-                    showResetVisibility()
-                }
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                DataTableScreen(
+                    tableName = tableName,
+                    navigateBack = {
+                        activity?.supportFragmentManager?.popBackStack()
+                    },
+                    onClick = {
+                        navigateToDatatableData(dataTable = it)
+                    }
+                )
             }
         }
-
-        return binding.root
-    }
-
-    override fun onRefresh() {
-        viewModel.loadDataTable(tableName)
-    }
-
-    private fun showUserInterface() {
-        setToolbarTitle(resources.getString(R.string.datatables))
-        val mLayoutManager = LinearLayoutManager(activity)
-        mLayoutManager.orientation = LinearLayoutManager.VERTICAL
-        binding.rvDataTable.layoutManager = mLayoutManager
-        binding.rvDataTable.setHasFixedSize(true)
-        binding.rvDataTable.adapter = dataTableAdapter
-        binding.swipeContainer.setColorSchemeColors(
-            *activity
-                ?.resources!!.getIntArray(R.array.swipeRefreshColors)
-        )
-        binding.swipeContainer.setOnRefreshListener(this)
-    }
-
-    private fun showDataTables(dataTables: List<DataTable>?) {
-        this.dataTables = dataTables
-        dataTableAdapter.dataTables = dataTables ?: emptyList()
     }
 
-    private fun showEmptyDataTables() {
-        binding.llError.visibility = View.VISIBLE
-        binding.rvDataTable.visibility = View.GONE
-        binding.tvError.setText(R.string.empty_data_table)
+    private fun navigateToDatatableData(dataTable: DataTable) {
+        val action =
+            DataTableFragmentDirections.actionDataTableFragmentToDataTableDataFragment(
+                dataTable,
+                entityId
+            )
+        findNavController().navigate(action)
     }
 
-    private fun showResetVisibility() {
-        binding.llError.visibility = View.GONE
-        binding.rvDataTable.visibility = View.VISIBLE
-    }
-
-    private fun showError(message: Int) {
-        Toaster.show(binding.root, message)
-        binding.llError.visibility = View.VISIBLE
-        binding.rvDataTable.visibility = View.GONE
-        binding.tvError.text = getString(R.string.failed_to_fetch_datatable)
-    }
-
-    private fun showProgressbar(show: Boolean) {
-        if (show && dataTableAdapter.itemCount == 0) {
-            binding.progressbarDataTable.visibility = View.VISIBLE
-            binding.swipeContainer.isRefreshing = false
-        } else {
-            binding.progressbarDataTable.visibility = View.GONE
-            binding.swipeContainer.isRefreshing = show
-        }
+    override fun onResume() {
+        super.onResume()
+        toolbar?.visibility = View.GONE
     }
```

Comment:
Similarly, on the stop callback, add a toolbar?visibility = View.VISIBLE. Then we don't need to set the toolbar in datatabledatafragment.